### ISSUE TITLE
getFieldset doesn't correctly proxy a nested class object

### ISSUE THREAD

**Issue by ngbrown** [edmundhung/conform#593]
### Describe the bug and the expected behavior

I have a form with initial values (`defaultValue`) and a [nested array](https://conform.guide/complex-structures#nested-array), but the `initialValue` was always `undefined`. It took me awhile to figure out what is going on, but it appears that it was because the nested object was a `class`. This is a simple class with just public properties.

### Conform version

v1.1.0

### Steps to Reproduce the Bug or Issue

View this snippet: https://stackblitz.com/edit/remix-run-remix-cym13r?file=app%2Froutes%2F_index.tsx

You can alternate between using a `class` and plain JavaScript objects. The location to access the `id` value changes between the two. It should be consistent.

### What browsers are you seeing the problem on?

Chrome, Firefox

### Additional context

The actual class of values is generated for `protobufjs`, but I reproduced it with just a simple `class` and `constructor`.

**Comment by edmundhung** [comment]
I don't think we can support a class instance in the default value unless it is serializable (This is not implemented, just an idea). Conform does an additional normalization step to ensure that the default value can be compared with the form value for additional checks such as the dirty state. But this won't work correctly in your case. 

You will need to convert it to an object literal before passing it to the default value. Conform should probably throw an error if it finds this as well.
